### PR TITLE
desktop app integration

### PR DESCRIPTION
**Pull request by ericxtang**
Integration with the desktop app

* Get a stream ID before streaming the video

### CODE REVIEW

File: livepeer/streaming/streamer.go [hunk 1]
```diff
@@ -223,13 +223,22 @@ func (self *Streamer) GetRTMPBuffer(id string) (buf av.Demuxer) {
 	return
 }
 
+func (self *Streamer) GetAllRTMPBufferIDs() []StreamID {
+	ids := make([]StreamID, 0, len(self.rtmpBuffers))
+	for k, _ := range self.hlsBuffers {
```
Comment (dob) [inline livepeer/streaming/streamer.go]: should this be `self.hlsBuffers` or `self.rtmpBuffers`?

File: livepeer/streaming/streamer.go [hunk 2]
```diff
@@ -325,23 +343,35 @@ func (self *Streamer) GetNetworkStream(id StreamID) *lpmsStream.VideoStream {
 	return self.networkStreams[id]
 }
 
+func (self *Streamer) GetAllNetworkStreams() []*lpmsStream.VideoStream {
```
Comment (dob) [inline livepeer/streaming/streamer.go:346]: What's a NetworkStream again?

**Review by dob — CHANGES_REQUESTED** [review]

File: livepeer/streaming/streamer.go (hunk 2 above)
Comment (ericxtang) [reply livepeer/streaming/streamer.go:346]: NetworkStream is the new stream concept.  We are phasing out of the old stream.  So now, streamer keeps track of "NetworkStreams", and they can be subscribed to via "HLSBuffer" or "RTMPBuffer".

**Review by dob — APPROVED** [review]
🚢

**Pull request merged**